apps/src/templates/studioHomepages/Courses.jsx

```diff
@@ -165,8 +161,24 @@ const Courses = React.createClass({
           </div>
         )}
 
-        {!(isEnglish && isTeacher) && (
-          <div>
+        {/* Signed-in student in English */}
+        {(!isSignedOut && !isTeacher && isEnglish) && (
+          <ContentContainer
+            heading={i18n.teacherCourseHoc()}
+            description={i18n.teacherCourseHocDescription()}
+            isRtl={isRtl}
+            linkText={i18n.teacherCourseHocLinkText()}
+            link={`${codeOrgUrlPrefix}/learn`}
+            showLink={true}
+          >
+            <ProtectedStatefulDiv ref="hocCoursesFirstRow"/>
+          </ContentContainer>
+        )}
+
+        {/* Anything but a teacher or student in English.
+            That is: signed-out, or a student or teacher in non-English. */}
+        {(isSignedOut || !isEnglish) && (
+          <div id="all-courses">
```

Comment:
When I'm reading JSX and I see an id, my first thought is "that means there are side effects somewhere else that I need to track down" (either CSS, or jquery doing something). To some extent it's conceptually similar to adding a global variable (although arguably not quite egregious). As such, I prefer avoiding them unless absolutely necessary.